docs(operators): updates to the operator overviews

PaulRMellor opened the pull request:
**Documentation**

Update and refactor of the operator overviews in the Overview and Deploying guide.
Incorporates Access Operator in the Overview and introduces the operators earlier in the guide, with separate sections describing each.
The operators overview in the Overview guide describes the operators
The operators overview in the Deploying guide is more deployment-specific.

As discussed for https://github.com/strimzi/kafka-access-operator/issues/71#issuecomment-2641401910


### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [ ] Make sure all tests pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md
- [ ] Supply screenshots for visual changes, such as Grafana dashboards

scholzj commented on documentation/assemblies/deploying/assembly-deploy-intro-operators.adoc:
```diff
@@ -7,27 +7,25 @@
+
+Additionally, Strimzi provides Drain Cleaner, which supports the Cluster Operator in managing pod evictions for Kafka clusters.
+
+The Cluster Operator is required and must be deployed first.
+It handles the deployment of Kafka and, if configured, can also deploy the Entity Operator, which may include the Topic Operator, User Operator, or both.
+
+The Topic Operator and User Operator can also be deployed standalone to manage topics and users for a Kafka cluster that is not managed by Strimzi.
```
If we want to list the Access Operator and Drain Cleaner in the introduction in sich a prominent way, we should probably also cover here how they are installed.

ppatierno replied:
+1

scholzj reviewed (commented):
TBH, for me the Access Operator does not fit here:
* The way you added it here is very forced.
* It fails to properly explain what it does, why and how to use it, and who should use it. This will just lead to many questions from confused people asking `Why do I need it?`. While in reality 99% of users will never need it.
* While the Access Operator has its first release, it is largely unfinished. The actual value will be added only once it has cross cluster support. Until then, copying the Secrets using existing tools is likely better choice.
* If you want to include the Access Operator more prominently in the docs, you should do it in the places which cover the use cases it handles. Such as in the detailed sections about User Operator and how it handles namespaces.

The https://github.com/strimzi/kafka-access-operator/issues/71 issue needs to be interpreted in the right context. Drain Cleaner is indeed something what everyone using Strimzi to run Kafka brokers should be aware of. And if users are not, that is something to improve. But the Access Operator is not - and will not be even once it is feature complete. It is a tool in the same level as for example the Configuration Providers - it has some use-cases and it has value for some users. But not everyone needs to be aware of it and not everyone needs it. So its presence in the documentation should correspond to it.

ppatierno commented on documentation/modules/overview/con-overview-components-access-operator.adoc:
```diff
@@ -0,0 +1,13 @@
+= Access Operator
+
+[role="_abstract"]
+If you are using the Cluster Operator to manage your Kafka cluster, you can deploy and use the Access Operator to manage the distribution of connection details 
```
```suggestion
If you are using the Cluster Operator to manage your Kafka cluster, you can deploy and use the Access Operator to manage the distribution of connection details.
```

PaulRMellor commented:
> TBH, for me the Access Operator does not fit here:
> 
> * The way you added it here is very forced.
> * It fails to properly explain what it does, why and how to use it, and who should use it. This will just lead to many questions from confused people asking `Why do I need it?`. While in reality 99% of users will never need it.
> * While the Access Operator has its first release, it is largely unfinished. The actual value will be added only once it has cross cluster support. Until then, copying the Secrets using existing tools is likely better choice.
> * If you want to include the Access Operator more prominently in the docs, you should do it in the places which cover the use cases it handles. Such as in the detailed sections about User Operator and how it handles namespaces.
> 
> The [strimzi/kafka-access-operator#71](https://github.com/strimzi/kafka-access-operator/issues/71) issue needs to be interpreted in the right context. Drain Cleaner is indeed something what everyone using Strimzi to run Kafka brokers should be aware of. And if users are not, that is something to improve. But the Access Operator is not - and will not be even once it is feature complete. It is a tool in the same level as for example the Configuration Providers - it has some use-cases and it has value for some users. But not everyone needs to be aware of it and not everyone needs it. So its presence in the documentation should correspond to it.

Thanks @scholzj. I see your point and the risk of confusion for users who don’t require the access operator, but I do think it belongs in the Strimzi Operators section for completeness. Would it help to adjust the description to make it clear that it's an advanced, optional tool with specific use cases? We can refine the explanation so that it’s not just listed but gives a clearer picture of when and why users might consider it. 

We can also look into adding more detail on covering multi-namespace setups, where its value becomes clearer.

PaulRMellor commented:
Following discussion, I reduced the profile of the access operator: removed from overview, minimal content in deploying guide.  I've also moved the access operator section in the deploying guide to sit under the section on Using the User Operator to manage Kafka users
